modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java

```diff
@@ -1164,6 +1202,35 @@ public void paint(final Graphics2D graphics, final GridCoverage2DReader gridCove
 
         setupInterpolationHints(interpolation);
         
+        //
+        // Band selection
+        //
+        int[] bandIndicesTmp = null;
+        ChannelSelection channelSelection = symbolizer.getChannelSelection();
```

Comment:
Does it make sense to do the band selection and check for the bands param support just once? Or at least externalize that in a single method? You are two copies grabbing the indices from the channel selection, here and above.